webapp/src/components/jira_ticket_tooltip/jira_ticket_tooltip.tsx

```diff
@@ -0,0 +1,312 @@
+                        className='popover-labels__label-list'
+                    >
+                        {moreLabels}
+                    </span>);
+            }
+
+            return <></>;
+        });
+        return (<div className='popover-labels__label'>{labelList}</div>);
+    }
+
+    render() {
+        if (!this.state.isLoaded) {
+            return (<p/>);
+        }
+
+        const {ticketDetails, href: jiraTicketURI} = this.state;
+
+        let jiraAvatar;
+        let jiraIssueIconURI;
+        let jiraTicketKey;
+        let jiraTicketTitle;
+        let jiraTicketAssigneeAvatarURI;
+        let jiraTicketAssigneeName;
+        let jiraTicketStatusName;
+        let jiraTicketDescription;
+        let jiraTicketVersions;
+        let jiraTicketLabels;
+        if (ticketDetails) {
+            jiraAvatar = ticketDetails.jiraIcon;
+            jiraIssueIconURI = ticketDetails.issueIcon;
+            jiraTicketKey = ticketDetails.ticketId;
+            jiraTicketTitle = ticketDetails.summary;
+            jiraTicketAssigneeAvatarURI = ticketDetails.assigneeAvatar;
+            jiraTicketAssigneeName = ticketDetails.assigneeName;
+            jiraTicketStatusName = ticketDetails.statusKey;
+            jiraTicketDescription = ticketDetails.description;
+            jiraTicketVersions = ticketDetails.versions;
+            jiraTicketLabels = ticketDetails.labels;
+        }
+
+        return (
+            <div className='ticket-popover'>
+                <div className='popover-header'>
+                    <div className='popover-header__container'>
+                        <a
+                            href={jiraTicketURI}
+                            title='Go to ticket'
+                        >
+                            <img
+                                src={jiraAvatar}
+                                width={14}
+                                height={14}
+                                alt='jira-avatar'
+                                className='popover-header__avatar'
+                            />
+                        </a>
+                        <a
+                            href={jiraTicketURI}
+                            className='popover-header__keyword'
+                        >
+                            <span className='jira-ticket-key'>{jiraTicketKey}</span>
+                            <img
+                                alt='jira-issue-icon'
+                                width='14'
+                                height='14'
+                                src={jiraIssueIconURI}
+                            />
+                        </a>
+                    </div>
+                </div>
+                <div className='popover-body'>
+                    <div className='popover-body__title'>
+                        <a href={jiraTicketURI}>
+                            <h5>{this.truncateString(jiraTicketTitle as string, jiraTicketTitleMaxLength)}</h5>
```

Comment:
Why do we need to cast this `as string`? I'm thinking we should let typescript take care of this by checking if `ticketDetails` is defined, and return `null` or some other short-circuiting return value so we don't need to cast things like this. All of the `as *` here are a little worrisome.

Reply:
Also what value does `truncateString` give us? Why not just `jiraTicketTitle.substring(0, jiraTicketTitleMaxLength)`